client/windows/CMapOverview.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * CMapOverview.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+
+#include "CMapOverview.h"
+
+#include "../lobby/SelectionTab.h"
+
+#include <vstd/DateUtils.h>
+
+#include "../gui/CGuiHandler.h"
+#include "../gui/WindowHandler.h"
+#include "../widgets/CComponent.h"
+#include "../widgets/MiscWidgets.h"
+#include "../widgets/TextControls.h"
+#include "../windows/GUIClasses.h"
+#include "../windows/InfoWindows.h"
+#include "../render/CAnimation.h"
+#include "../render/Canvas.h"
+#include "../render/IImage.h"
+#include "../render/IRenderHandler.h"
+#include "../render/Graphics.h"
+
+#include "../../lib/CGeneralTextHandler.h"
+#include "../../lib/CConfigHandler.h"
+#include "../../lib/campaign/CampaignState.h"
+#include "../../lib/mapping/CMap.h"
+#include "../../lib/mapping/CMapService.h"
+#include "../../lib/mapping/CMapInfo.h"
+#include "../../lib/mapping/CMapHeader.h"
+#include "../../lib/mapping/MapFormat.h"
+#include "../../lib/TerrainHandler.h"
+#include "../../lib/filesystem/Filesystem.h"
+
+#include "../../lib/serializer/BinaryDeserializer.h"
+#include "../../lib/StartInfo.h"
+#include "../../lib/rmg/CMapGenOptions.h"
+
+CMapOverview::CMapOverview(std::string mapName, std::string fileName, std::string date, ResourcePath resource, ESelectionScreen tabType)
+	: CWindowObject(BORDERED | RCLICK_POPUP), resource(resource), mapName(mapName), fileName(fileName), date(date), tabType(tabType)
+{
+
+	OBJ_CONSTRUCTION_CAPTURING_ALL_NO_DISPOSE;
+
+	const JsonNode config(JsonPath::builtin("config/widgets/mapOverview.json"));
+	pos = Rect(0, 0, config["items"][0]["rect"]["w"].Integer(), config["items"][0]["rect"]["h"].Integer());
+
+	widget = std::make_shared<CMapOverviewWidget>(*this);
```

Comment:
Why not something like this?
```suggestion
	widget = std::make_shared<CMapOverviewWidget>(*this);
    pos = widget->pos;
```

Reply:
Tested this. But don't work.
The border then think the window is 0x0px and is totally destroyed.

Reply:
Yes, because your need to take imageArea, not pos from FilledTexture

Reply:
`imageArea` is not accessable (protected)... But `pos` from texture is working.

Reply:
then maybe introduce getImageArea()

Reply:
Or finally fix Configurable UI and initialize pos as part of UI loading

Reply:
I think it should be okay like it is now!?

Reply:
yes, let's merge

Reply:
there is a conflict

Reply:
conflict resolved